### PR TITLE
Security: Prevent arbitrary file write and read

### PR DESCRIPTION
**Pull request by SchoolGuy**
- Prevent arbitrary file write through upload_log_data
- Prevent arbitrary file read through generate_script
- Prevent log poisoning
- Check if modify_setting is really allowed (and adjust tests)

This is a backport of PR #2794
Git-SHA of Merge Commit on master: d8f60bbf

### CODE REVIEW

File: cobbler/remote.py [hunk 1]
```diff
@@ -2381,58 +2399,73 @@ def upload_log_data(self, sys_name, file, size, offset, data, token=None, **rest
...
-        if fn.startswith('..'):
-            raise CX("invalid filename used: %s" % fn)
-
-        # FIXME ... get the base dir from cobbler settings()
-        udir = "/var/log/cobbler/anamon/%s" % sys_name
-        if not os.path.isdir(udir):
-            os.mkdir(udir, 0o755)
-
-        fn = "%s/%s" % (udir, fn)
+        contents = base64.decodebytes(data)
```
Comment (vzhestkov) [inline cobbler/remote.py:2442]: `decodebytes` was appended in 3.1, do we need the support of Python 2.7 for this version of Cobbler?

Comment (SchoolGuy) [reply cobbler/remote.py:2442]: Python 2.7 was not supported since 3.0.0, so no we don't need to support it.

**Review by agraul — APPROVED** [review]
LGTM

**Review by vzhestkov — APPROVED** [review]
Looks good

**Review by meaksh — APPROVED** [review]
LGTM

**Pull request merged**